pkg/serviceprovider/snyk/snyk.go

```diff
@@ -0,0 +1,117 @@
+package snyk
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	api "github.com/redhat-appstudio/service-provider-integration-operator/api/v1beta1"
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/serviceprovider"
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/config"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var _ serviceprovider.ServiceProvider = (*Snyk)(nil)
+
+type Snyk struct {
+	Configuration config.Configuration
+	lookup        serviceprovider.GenericLookup
+	httpClient    rest.HTTPClient
+}
+
+var Initializer = serviceprovider.Initializer{
+	Probe:       snykProbe{},
+	Constructor: serviceprovider.ConstructorFunc(newSnyk),
+}
+
+func newSnyk(factory *serviceprovider.Factory, _ string) (serviceprovider.ServiceProvider, error) {
+
+	cache := serviceprovider.NewMetadataCache(factory.KubernetesClient, &serviceprovider.NeverMetadataExpirationPolicy{})
+
+	return &Snyk{
+		Configuration: factory.Configuration,
+		lookup: serviceprovider.GenericLookup{
+			ServiceProviderType: api.ServiceProviderTypeSnyk,
+			TokenFilter:         &tokenFilter{},
+			MetadataProvider: &metadataProvider{
+				httpClient:   factory.HttpClient,
+				tokenStorage: factory.TokenStorage,
+			},
+			MetadataCache: &cache,
+			RepoHostParser: serviceprovider.RepoHostParserFunc(func(repoUrl string) (string, error) {
+				return repoUrl, nil
+			}),
+		},
+		httpClient: factory.HttpClient,
+	}, nil
+}
+
+var _ serviceprovider.ConstructorFunc = newSnyk
+
+func (g *Snyk) GetOAuthEndpoint() string {
+	return ""
+}
+
+func (g *Snyk) GetBaseUrl() string {
+	return "https://snyk.io"
+}
+
+func (g *Snyk) GetType() api.ServiceProviderType {
+	return api.ServiceProviderTypeSnyk
+}
+
+func (g *Snyk) TranslateToScopes(_ api.Permission) []string {
+	return []string{}
+}
+
+func (g *Snyk) LookupToken(ctx context.Context, cl client.Client, binding *api.SPIAccessTokenBinding) (*api.SPIAccessToken, error) {
+	tokens, err := g.lookup.Lookup(ctx, cl, binding)
```

Comment:
This doesn't really have to call the generic look up actually. If we ever are going to support one token per namespace.. You can just `cl.List()` the first token labeled with the snyk host and return that...

But I guess that's what generic token lookup does anyway so maybe it is good to leave this code like it is.